### ISSUE TITLE
dynalink in nashorn causing memory leak

### ISSUE THREAD

**Issue by subijayb** [szegedi/dynalink#15]
Hi @szegedi ,

We have an enterprise application running on wildfly and using open JDK 11.0.22. We use Camunda for business process flow and Camunda uses Nashorn. In perf test we see a leak. Upon research we suspect our problem is same as 
https://stackoverflow.com/questions/62593886/java-memory-leak-why-are-unreachable-objects-not-collected 
This in turn takes us to JDK bug https://bugs.openjdk.org/browse/JDK-8229011

I know, Nashhorn is no longer part of open jdk since 15. We picked up the lib from github project https://github.com/openjdk/nashorn and we are using this as we are currently moving to use JDK 17.

So our question is if the bug raised in OpenJDK is still present in the latest release of nashorn 15.4? If yes, is there a plan to fix this

P.S. I did not see 'Issues' section under https://github.com/openjdk/nashorn. Hence posting this question here.

**Comment by szegedi** [comment]
Hi @subijayb there's a known [memory leak in Dynalink](https://github.com/openjdk/jdk/pull/1918) that was fixed in Java 16. This could be the same. Can you re-test your app with Java 16 or later (e.g. Java 17 you want to move to) and let me know if the problem still persists?

**Comment by subijayb** [comment]
Thanks @szegedi for the quick reply.
It will take a little bit more time for our application to fully run on JDK 17 as we are doing jakarta upgrade also :)

Meanwhile, I tested with the reproducer attached in [JDK-8229011](https://bugs.openjdk.org/browse/JDK-8229011), against 11 and 17. In 11 I see same stack in the hapdump once I import in MAT. But when I run on 17, the heap dump is clean. 
I think this should fix our problem. Thanks !!

One change I did in the reproducer code to make it run on 17 is to use

```
final ScriptEngine scriptEngine = new org.openjdk.nashorn.api.scripting.NashornScriptEngineFactory()
      .getScriptEngine();
```

instead of

`final ScriptEngine scriptEngine = new ScriptEngineManager().getEngineByName("nashorn");`
